generator/src/generator/namespace.rs

```diff
@@ -3378,24 +3438,42 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                     if let Some(list_length) = list_field.length() {
                         outln!(
                             out,
-                            "let ({}, remaining) = crate::x11_utils::parse_u8_list(remaining, \
+                            "let ({}, remaining) = crate::x11_utils::parse_u8_list({}, \
                              {})?;",
                             rust_field_name,
+                            from,
                             list_length,
                         );
-                        outln!(
-                            out,
-                            "let {} = <[u8; {}]>::try_from({}).unwrap();",
-                            rust_field_name,
-                            list_length,
-                            rust_field_name,
-                        );
+                        if container == FieldContainer::Other {
+                            // Only force taking ownership for non-request types.
+                            outln!(
+                                out,
+                                "let {} = <[u8; {}]>::try_from({}).unwrap();",
+                                rust_field_name,
+                                list_length,
+                                rust_field_name,
+                            );
+                        } else {
+                            // Otherwise convert this into a reference to a fixed length array.
+                            // It's basically dumb luck that the largest list in the X11 protocol
+                            // and the largest fixed length array that Rust has a builtin conversion
+                            // for going from &[T] to &[T; N] both happen to be 32.
+                            assert!(list_length <= 32);
+                            outln!(
+                                out,
+                                "let {} = <&[u8; {}]>::try_from({}).unwrap();",
+                                rust_field_name,
+                                list_length,
+                                rust_field_name,
```

Comment:
I was going to say that the code in the `if` and the `else` are identical. After lots of staring, I figured out that there is a "one `&`" difference. Does it make sense to factor this out with something like `let is_ref = if container == Other { "" } else { assert!; "&" };`? (pseudo-code, but you get the idea)
(Alternatively: `true`/`false` instead of `""`/"&"`)

Reply:
Done.